refactor(diagnostic): set sign by using extmark (#26193)
neovim/neovim: runtime/doc/news.txt

• |vim.wait()| is no longer allowed to be called in |api-fast|.

> • Lsp diagnostic sign texts defined via |sign_define()| no longer work.
>   Configure it by using |vim.diagnostic.config()|.

Reviewer: Not specific to LSP

```suggestion
• Diagnostic sign text is no longer configured with |sign_define()|.
  Use |vim.diagnostic.config()| instead.
```

• Vimscript function |exists()| supports checking |v:lua| functions.

• Diagnostic sign text is no longer configured with |sign_define()|.

>   Use |vim.diagnostic.config()| instead.

Reviewer: This should be in the breaking changes section.
